tests/integration-tests/tests/common/utils.py

```diff
@@ -401,3 +401,48 @@ def wait_process_completion(remote_command_executor, pid):
         raise Exception("The process is still running")
     else:
         return result.stdout.strip()
+
+
+def get_deployed_config_version(cluster, compute_node_ip: str = None):
+    """Retrieves the cluster config version deployed on the cluster node from its dna.json
+    If 'compute_node_ip' is specified, the config version will be retrieved from the compute node;
+    otherwise, it will be retrieved from the head node.
+    """
+    dna_json = get_deployed_dna_json(cluster, compute_node_ip)
+
+    return dna_json["cluster"]["cluster_config_version"]
+
+
+def get_deployed_dna_json(cluster, compute_node_ip: str = None):
+    """Retrieves the dna.json from the cluster node
+    If 'compute_node_ip' is specified, it will be retrieved from the compute node;
+    otherwise, it will be retrieved from the head node.
+    """
+    command = "sudo cat /etc/chef/dna2.json"
```

Comment:
why the name is "dna2.json"? This does not seem a standard name. If it's a test specific name, let's pass it as parameter to the function with a default value "dna.json"

Reply:
typo. Made locally to temporarily test an unhappy path. Fixed in next rev.